donkeycar/parts/lidar.py

```diff
@@ -13,6 +14,187 @@
 from donkeycar.utils import norm_deg, dist, deg2rad, arr_to_img
 from PIL import Image, ImageDraw
 
+
+CLOCKWISE = 1
+COUNTER_CLOCKWISE = -1
+
+
+class RPLidar2(object):
+    '''
+    Adapted from https://github.com/Ezward/rplidar
+    '''
+    def __init__(self,
+                 min_angle = 0.0, max_angle = 360.0,
+                 min_distance = sys.float_info.min, max_distance = sys.float_info.max,
+                 forward_angle = 0.0,
+                 angle_direction=CLOCKWISE,
+                 debug=False):
+        
+        self.lidar = None
+        self.port = None
+        self.on = False
+        
+
+        help = []
+        if min_distance < 0:
+            help.append("min_distance must be >= 0")
+
+        if max_distance <= 0:
+            help.append("max_distance must be > 0")
+            
+        if min_angle < 0 or min_angle > 360:
+            help.append("min_angle must be 0 <= min_angle <= 360")
+
+        if max_angle <= 0 or max_angle > 360:
+            help.append("max_angle must be 0 < max_angle <= 360")
+          
+        if forward_angle < 0 or forward_angle > 360:
+            help.append("forward_angle must be 0 <= forward_angle <= 360")
+            
+        if angle_direction != CLOCKWISE and angle_direction != COUNTER_CLOCKWISE:
+            help.append("angle-direction must be 1 (clockwise) or -1 (counter-clockwise)")
+
+        if len(help) > 0:
+            msg = "Could not start RPLidar; bad parameters passed to constructor"
+            print(msg)
```

Comment:
We moved from printing to logging. - This is a nice way to assemble a comprehensive help message, very useful!

Reply:
Thanks.  I switch to logging everywhere except the main, which is using console io.